src/components/ADempiere/TabParent/index.vue

```diff
@@ -0,0 +1,182 @@
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <https:www.gnu.org/licenses/>.
+-->
+
+<template>
+  <el-tabs
+    v-model="currentTab"
+    type="border-card"
+    :before-leave="handleBeforeLeave"
+    @tab-click="handleClick"
+  >
+    <template v-for="(tabAttributes, key) in tabsList">
+      <el-tab-pane
+        :key="key"
+        :label="tabAttributes.name"
+        :windowuuid="windowUuid"
+        :tabuuid="tabAttributes.uuid"
+        :name="String(key)"
+        lazy
+        :disabled="isDisabledTab(key)"
+        :style="tabParentStyle"
+      >
+        <lock-record
+          slot="label"
+          :tab-position="key"
+          :tab-uuid="tabAttributes.uuid"
+          :table-name="tabAttributes.tableName"
+          :tab-name="tabAttributes.name"
+        />
+
+        <panel-definition
+          :parent-uuid="windowUuid"
+          :container-uuid="tabAttributes.uuid"
+          :group-tab="tabAttributes.tabGroup"
+          :panel-type="panelType"
+        />
+      </el-tab-pane>
+    </template>
+  </el-tabs>
+</template>
+
+<script>
+import { defineComponent, computed, ref, watch } from '@vue/composition-api'
+
+import PanelDefinition from '@/components/ADempiere/PanelDefinition'
+import LockRecord from '@/components/ADempiere/ContainerOptions/LockRecord'
+import { parseContext } from '@/utils/ADempiere/contextUtils'
+
+export default defineComponent({
+  name: 'TabParent',
+
+  components: {
+    PanelDefinition,
+    LockRecord
+  },
+
+  props: {
+    windowUuid: {
+      type: String,
+      default: ''
+    },
+    windowMetadata: {
+      type: Object,
+      default: () => {}
+    },
+    tabsList: {
+      type: Array,
+      default: () => []
+    }
+  },
+
+  setup(props, { root }) {
+    const panelType = 'window'
+    const currentTab = ref(root.$route.query.tabParent)
+    const tabUuid = ref(props.tabsList[0].uuid)
+
+    const tabParentStyle = computed(() => {
+      // if tabs children is showed or closed
+      if (props.windowMetadata.isShowedTabsChildren) {
+        return {
+          height: '100%',
+          overflow: 'hidden'
+        }
+      }
+      return {
+        height: '75vh',
+        overflow: 'auto'
+      }
+    })
+
+    const isCreateNew = computed(() => {
+      return Boolean(root.$route.query.action === 'create-new')
+    })
+
+    const isDisabledTab = (key) => {
+      return key > 0 && isCreateNew.value
+    }
+
+    const setCurrentTab = () => {
+      root.$store.dispatch('setCurrentTab', {
+        parentUuid: props.windowUuid,
+        containerUuid: tabUuid.value,
+        window: props.windowMetadata
+      })
+    }
+
+    const handleBeforeLeave = (activeName) => {
```

Comment:
What is the reason for it? Handle before leave?

Reply:
Delete unused methods